ang/volunteer/Project.js

```diff
@@ -91,20 +82,29 @@
       if (CRM.vars['org.civicrm.volunteer'].entityTitle) {
         project.title = CRM.vars['org.civicrm.volunteer'].entityTitle;
       }
-      //VOL-223
-      $scope.editContactType = supporting_data.values.canEditContacts;
     } else {
-      //VOL-223
-      $scope.editContactType = can_edit_contact_types.values;
       $(relationship_data.values).each(function (index, relationship) {
-        if (!relationships.hasOwnProperty(relationship.relationship_type_id)) {
-          relationships[relationship.relationship_type_id] = [];
+        if (!volRelData.hasOwnProperty(relationship.relationship_type_id)) {
+          volRelData[relationship.relationship_type_id] = [];
         }
-        relationships[relationship.relationship_type_id].push(relationship.contact_id);
+        volRelData[relationship.relationship_type_id].push({
+          contact_id: relationship.contact_id,
+          can_be_read_by_current_user: relationship.can_be_read_by_current_user
+        });
       });
-      var originalRelationships = _.clone(relationships);
     }
+
+    // flatten the data a bit to make it easier to work with in the template
+    _.each(volRelData, function (contacts, relTypeId) {
+      relationships[relTypeId] = [];
+      showRelationshipType[relTypeId] = false;
+      _.each(contacts, function (contact) {
+        relationships[relTypeId].push(contact.contact_id);
+        showRelationshipType[relTypeId] = showRelationshipType[relTypeId] || contact.can_be_read_by_current_user;
```

Comment:
You used an OR here, where I had used an AND. This is a design decision we should come to an agreement on and document well. If we use OR, that means we show the widget if they can view any one of the related contacts. If they are missing one, that person would be stripped out of the project and the person editing it would never know.
If we use an AND, they have to have access to all of the contacts, and if they don't they can't edit the relationship contacts at all. I felt like the latter was a better choice.